hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -465,9 +495,14 @@ protected void finalize() throws Throwable {
     super.finalize();
     if (!closed) {
       close(); // releasing resources is more important than printing the warning
-      String trace = Joiner.on("\n\t").join(
+      String trace = Joiner.on("\\n\t").join(
```

Comment:
Is the additional `\` here intentional? If so, what behaviour changes?

Reply:
Collapsing the multiline log to a single line - was not brave enough to remove the \n-s altogether.
This should create logs like:
```
Unclosed input stream created by:\n {Stacktrace.line1\n  Stractrace.line2\n  Stractrace.line3}
```

Reply:
This makes logs harder to read. What is the intended purpose of this change?

Reply:
When I have to read Hive logs where there are plenty of concurrent queries running parallel, extra log lines will appear inside if we have multiline logs which will make them hard to read (even harder than the one proposed above).
Also only the first line of the log will get the timestamp and other log header information, which will make it hard to use any log analyzer tool to check them.

These are the main reasons in Hive we usually try to avoid multiline logs, and accept the drawback of the "hardness" of reading them.
I am open to suggestions, even reverting back to this change if you strongly feel that it should remain as it is.

Reply:
Reverted in the meantime to proceed with the patch

Reply:
@pvary, I'm not sure why, but I've never seen mixed log lines in our environment. So I don't think it is a good idea to solve the problem for one environment by making logs harder to read everywhere.

What about using a custom appender or SLF4J binding in your environment that converts newlines to `\\n` so that you can configure it locally? I think that is a fairly easy solution that doesn't require changing logs everywhere.

Reply:
Ok. Keeping the reverted stuff